perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/impl/RegistrarManagerImpl.java

```diff
@@ -3925,8 +3925,10 @@ private int processApplication(PerunSession session, Application application, Li
 			}
 		}
 
-		// store user-ext-source attributes from session to application object
+		// store user-ext-source attributes and redirectURL from session to application object
 		LinkedHashMap<String, String> map = new LinkedHashMap<>(session.getPerunPrincipal().getAdditionalInformations());
+		String redirectURL = BeansUtils.stringToMapOfAttributes(application.getFedInfo()).get("redirectURL");
```

Comment:
This is wrong. The method `stringToMapOfAttributes` is used for a completely different purpose. 

The fedInfo has a JSON format, something like this: 
```
{ displayName="Vojta Petr von Sassmann" commonName="null" givenName="Vojta" sureName="von Sassmann" loa="null" mail="sdf@sdf.sd" organization="true" redirectURL="localhost/krb/registrar/?group=group" }
```

From such a string, you need to parse the redirectURL. You can probably use some Json parsing library and use it for this purpose. 

Also, please make these changes as soon as possible, we need this functionality to be finished by the end of this week.